tests/e2e/restart_test.go

```diff
@@ -84,7 +84,7 @@ func (s *RestartTestData) TestVolumeDetached(c *C) {
 	c.Assert(err, IsNil, Commentf(out))
 
 	// 4. Verify detached status. Volume should be detached (within the timeout)
-	status = verification.VerifyDetachedStatus(s.volumeName, s.config.DockerHosts[1], s.config.EsxHost)
+	status = verification.PollDetachedStatus(s.volumeName, s.config.DockerHosts[1], s.config.EsxHost)
```

Comment:
The test code is getting really complex with different functions with the same purpose doing things differently. Test code development and reviews becomes that much more tricky with having to use a particular function (for the same purpose) in different scenarios. In this case all that a test code needs to figure is if a volume is detached, correct? For that a function which checks the current state and returns if its detached or wait  if the state must become detached. A single API that takes a waitTillVolumeDetached parameter is sufficient to embed both requirements if that's needed (still don't see why a function just wants the current state which need not be correct if its coming from the KV). Its simpler to have a single function used in all the test code.